salt/srv/salt/_modules/ceph.py

```diff
@@ -397,6 +397,37 @@ def get_cluster_object(cluster_name, sync_type, since):
             assert ret == 0
             data['crush'] = json.loads(raw)
 
+            # Inject DNE OSDs into the map to avoid inconsistencies between
```

Comment:
I think it may not be wise to modify the OSD map in this way: every other piece of code handling it would have to be aware of it being subtly different from the real osd map.

Injecting this kind of thing will not avoid the need for special handling elsewhere, as any code that e.g. reads uuid/public_addr/state would need modifying to cope with the "DNE" or null values injected here.

Given that, I think it's better to take the hit of handling this higher up the stack: any code looking up an OSD from ServerMonitor or looking it up based on the CRUSH map should explicitly handle the case where the OSD doesn't exist.